### PR TITLE
NOISSUE: CAS security example is very out of date

### PR DESCRIPTION
**Pull request by gregorydlogan**
This patch fixes two issues with the existing CAS security example file:

1) It adds a number of missing paths which were added to the default file, but not to the example
2) It attempts to re-enable CAS, which appears to have been accidentally removed.

### CODE REVIEW

**Comment by gregorydlogan** [comment]
Per discussion at https://groups.google.com/a/opencast.org/d/msg/users/NoZFBjxkzzg/aSa9t6UqBAAJ, this appears to be working for at least one user.

**Review by miesgre — COMMENTED** [review]
It does not work for me. I get this error:
```
2019-09-16 11:14:03,154 | ERROR | (SpringSecurityConfigurationArtifactInstaller:102) - Unable to refresh spring security configuration file /opencast-miesgre/build/opencast-dist-develop-7-SNAPSHOT/etc/security/mh_default_org.xml: {}
org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Filter beans '<casFilter>' and '<org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter#0>' have the same 'order' value. When using custom filters, please make sure the positions do not conflict with default filters. Alternatively you can disable the default filters by removing the corresponding child elements from <http> and avoiding the use of <http auto-config='true'>.
Offending resource: OSGi resource[file:/opencast-miesgre/build/opencast-dist-develop-7-SNAPSHOT/etc/security/mh_default_org.xml|bnd.id=113|bnd.sym=opencast-kernel]
```

File: etc/security/security_sample_cas.xml-example [hunk 1]
```diff
@@ -68,9 +310,21 @@
     <!-- Uncomment to enable x509 client certificates for identifying clients -->
     <!-- sec:x509 subject-principal-regex="CN=(.*?)," user-service-ref="userDetailsService" / -->
 
+    <!-- Enable and configure the failure URL for form-based logins -->
+    <sec:form-login authentication-failure-url="/admin-ng/login.html?error" authentication-success-handler-ref="authSuccessHandler" />
```
Comment (miesgre) [inline etc/security/security_sample_cas.xml-example]: Comment this line to make CAS work

**Review by miesgre — CHANGES_REQUESTED** [review]
It does not work for me. I get this error:
```
2019-09-16 11:14:03,154 | ERROR | (SpringSecurityConfigurationArtifactInstaller:102) - Unable to refresh spring security configuration file /opencast-miesgre/build/opencast-dist-develop-7-SNAPSHOT/etc/security/mh_default_org.xml: {}
org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: Filter beans '<casFilter>' and '<org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter#0>' have the same 'order' value. When using custom filters, please make sure the positions do not conflict with default filters. Alternatively you can disable the default filters by removing the corresponding child elements from <http> and avoiding the use of <http auto-config='true'>.
Offending resource: OSGi resource[file:/opencast-miesgre/build/opencast-dist-develop-7-SNAPSHOT/etc/security/mh_default_org.xml|bnd.id=113|bnd.sym=opencast-kernel]
```

**Comment by gregorydlogan** [comment]
@miesgre this should be working now!

**Comment by miesgre** [comment]
@gregorydlogan It works, but there are two issues:

1.  I get this warning. I don't know if it's important or not:
```
2019-09-17 12:04:53,826 | WARN  | (SingleSignOutHandler:147) - Front Channel single sign out redirects are disabled when the 'casServerUrlPrefix' value is not set.
```
2.  You need to configure a role provider to get roles, if not you get a bad credentials, and user is not logged in. This happens in the previous versions too (6.x, ...)

**Comment by gregorydlogan** [comment]
@miesgre no idea, I don't have anything to test against, just attempting to revert the merge by sight.

Regarding the role provider, that's expected isn't it?

**Review by miesgre — APPROVED** [review]
@gregorydlogan I think it can be merged as is.
I can Log in and not noticed any problems. It works in the same way old releases works

**Pull request merged**